mediabridge/main.py

```diff
@@ -1,31 +1,49 @@
 import logging
+import os
+from datetime import datetime
 
 import typer as typer
 from tqdm.contrib.logging import logging_redirect_tqdm
 
 from mediabridge.data_processing import wiki_to_netflix
 
 
-def main(verbose: bool = typer.Option(False, "--verbose", "-v"), test: bool = False):
-    if test:
+def main(
+    verbose: bool = typer.Option(False, "--verbose", "-v"),
+    log: bool = False,
+    test: bool = False,
+):
+    # create output directory if it doesn't exist
+    if not os.path.exists("./out"):
+        logging.warning(" /out directory does not exist, creating...")
+        os.mkdir("./out")
+
+    if log:
         # log all messages to file
         logging.basicConfig(
             level=logging.DEBUG,
-            filename="out/mediabridge.log",
-            filemode="a",
+            filename=f"out/mb_{datetime.now().strftime('%Y%m%d_%H%M%S')}.log",
+            filemode="w",
             format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
             datefmt="%H:%M:%S",
         )
-        wiki_to_netflix.process_data(True)
+        try:
+            wiki_to_netflix.process_data(test)
```

Comment:
We shouldn't call `process_data` from two different paths. There should be one place when we call it, which increases readability and future maintainability (what if we want to change something about the call and we forget to do it in both places?).

Instead, you should always catch the exception, and only log it if `log` is `True`.

I guess I missed this in the previous PR, sorry.

Reply:
My intention was to keep the calls separated so that we only use the tqdm handler when we're logging to stdout, but I don't think the handler has an impact on perf anyway so I consolidated it all in the latest commit.